testsuite/MDAnalysisTests/core/test_groups.py

```diff
@@ -352,46 +426,122 @@ def test_atomgroup_to_atomgroup(self, u):
         atm = u.atoms.atoms
         assert len(atm) == 125
         assert isinstance(atm, groups.AtomGroup)
+        assert atm is u.atoms
 
     def test_atomgroup_to_residuegroup(self, u):
-        res = u.atoms.residues
+        atm = u.atoms
+        res = atm.residues
         assert len(res) == 25
         assert isinstance(res, groups.ResidueGroup)
+        assert res == u.residues
+        assert res is not u.residues
+        assert res._cache['isunique'] is True
+        assert res._cache['unique'] is res
 
     def test_atomgroup_to_segmentgroup(self, u):
         seg = u.atoms.segments
         assert len(seg) == 5
         assert isinstance(seg, groups.SegmentGroup)
+        assert seg == u.segments
+        assert seg is not u.segments
+        assert seg._cache['isunique'] is True
+        assert seg._cache['unique'] is seg
 
     def test_residuegroup_to_atomgroup(self, u):
-        atm = u.residues.atoms
+        res = u.residues
+        atm = res.atoms
         assert len(atm) == 125
         assert isinstance(atm, groups.AtomGroup)
+        assert atm == u.atoms
+        assert atm is not u.atoms
+        # clear res' uniqueness caches:
+        if 'unique' in res._cache.keys():
+            del res._cache['unique']
+        if 'isunique' in res._cache.keys():
+            del res._cache['isunique']
+        atm = res.atoms
+        # assert uniqueness caches of atm are empty:
+        with pytest.raises(KeyError):
+            _ = atm._cache['isunique']
+        with pytest.raises(KeyError):
+            _ = atm._cache['unique']
+        # populate uniqueness cache of res:
+        assert res.isunique
+        atm = res.atoms
+        # assert uniqueness caches of atm are set:
+        assert atm._cache['isunique'] is True
+        assert atm._cache['unique'] is atm
 
     def test_residuegroup_to_residuegroup(self, u):
         res = u.residues.residues
         assert len(res) == 25
         assert isinstance(res, groups.ResidueGroup)
+        assert res is u.residues
```

Comment:
@orbeckst equivalent to `assert rg.residues is rg`

Reply:
ok, will change my commit